test/e2e/ingress-node-firewall/ingress-node-firewall.go

```diff
@@ -79,94 +65,54 @@ func loadFromFile(obj interface{}, fileName string) error {
-			}
-		}
-	}
-	return res
+func CreateIngressNodeFirewall(client *testclient.ClientSet, inf *ingressnodefwv1alpha1.IngressNodeFirewall,
+	timeout time.Duration) error {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	// we want to return failure if it already exists to avoid leaks between test cases.
+	return client.Create(ctx, inf)
 }
 
-func GetRuleCIDR(isSingleStack bool, nodes []v1.Node) (string, string, error) {
-	var v4CIDR, v6CIDR string
-	v4CIDRLen := 12
-	v6CIDRLen := 64
-	ips := NodesIPs(nodes)
-	for _, ip := range ips {
-		addr := net.ParseIP(ip)
-		if addr.To4() != nil {
-			v4Mask := net.CIDRMask(v4CIDRLen, 32)
-			v4CIDR = fmt.Sprintf("%s/%d", addr.Mask(v4Mask), v4CIDRLen)
-		} else if addr.To16() != nil {
-			v6Mask := net.CIDRMask(v6CIDRLen, 128)
-			v6CIDR = fmt.Sprintf("%s/%d", addr.Mask(v6Mask), v6CIDRLen)
-		} else {
-			return "", "", fmt.Errorf("invalid ip address family %s", ip)
-		}
-		if isSingleStack && (v4CIDR != "" || v6CIDR != "") {
-			break
-		}
-		if v4CIDR != "" && v6CIDR != "" {
-			break
-		}
-	}
-	return v4CIDR, v6CIDR, nil
+func DeleteIngressNodeFirewall(client *testclient.ClientSet, inf *ingressnodefwv1alpha1.IngressNodeFirewall,
+	timeout time.Duration) error {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	// we want to return failure if it already deleted to highlight possible test setup failure.
+	return client.Delete(ctx, inf)
 }
 
-func RunPingTest(nodes []v1.Node) (error, int) {
-	var errs []error
-	ii := NodesIPs(nodes)
-	for _, ip := range ii {
-		if _, err := exec.Command("ping", "-c", "1", ip).CombinedOutput(); err != nil {
-			errs = append(errs, err)
-		}
+func DeleteIngressNodeFirewallsWithLabels(client *testclient.ClientSet, namespace, label string, timeout time.Duration) error {
+	var infList *ingressnodefwv1alpha1.IngressNodeFirewallList
+	err := client.List(context.Background(), infList, goclient.InNamespace(namespace), goclient.HasLabels{label})
+	if err != nil {
+		return err
 	}
-	return kerrors.NewAggregate(errs), len(errs)
-}
-
-func GetDaemonSetPods(ns string) (*v1.PodList, error) {
-	var podList *v1.PodList
-	err := wait.PollImmediate(1*time.Second, 10*time.Second, func() (done bool, err error) {
-		podList, err = testclient.Client.Pods(ns).List(context.TODO(), metav1.ListOptions{
-			LabelSelector: "app=ingress-node-firewall-daemon",
-		})
-
-		if err != nil {
-			return false, err
-		}
-
-		if len(podList.Items) > 0 {
-			return true, nil
+	for _, inf := range infList.Items {
+		if err = DeleteIngressNodeFirewall(client, &inf, timeout); err != nil {
+			if !errors.IsNotFound(err) {
+				return err
+			}
 		}
-		return false, nil
-	})
-	return podList, err
-}
-
-func GetINF(operatorNamespace string, name string) *ingressnodefwv1alpha1.IngressNodeFirewall {
-	inf := &ingressnodefwv1alpha1.IngressNodeFirewall{}
-	inf.SetName(name)
-	inf.SetNamespace(operatorNamespace)
-	return inf
+	}
```

Comment:
we should check/wait also nodeState object to be deleted, similar to how we do currently

Reply:
I am not familiar with the code surrounding this. Isnt it deleted when no ingress node firewall config no longer targets a node? If so, why would we wait for it to be deleted when we are just deleting INFs?